### PR TITLE
Add light mode option

### PR DESCRIPTION
**Pull request by Gerrit0**
It is very bright outside my window today, yay snow, which makes it surprisingly difficult to see any dark mode site.

<details><summary>Dark mode (default)</summary>

</details>

<details><summary>Light mode</summary>

</details>

### CODE REVIEW

File: shared/src/types.ts [hunk 1]
```diff
@@ -2,3 +2,8 @@ export enum TreeMode {
   forEachChild,
   getChildren,
 }
+
+export enum Theme {
+  Dark = "dark",
+  Light = "light",
```
Comment (dsherret) [inline shared/src/types.ts:8]: Would be neat to add a "system" setting in the future maybe.

**Review by dsherret — APPROVED** [review]
Very nice. Thanks!

I mostly use light mode because I get eye strain if I use dark mode all day.

File: shared/src/types.ts (hunk 1 above)
Comment (Gerrit0) [reply shared/src/types.ts:8]: I'm up for doing that, will give it a shot

Comment (Gerrit0) [reply shared/src/types.ts:9]: Done! First time I've messed with `useReducer`, but it seemed like the right tool for the job in this case...

File: site/src/AppContext.tsx [hunk 2]
```diff
@@ -33,8 +34,14 @@ export function AppContextProvider({ children }: { children: React.ReactNode })
...
       showInternals: stateSaver.get().showInternals,
+      theme: stateSaver.get().theme,
     },
     compiler: undefined,
+    editorTheme: deriveEditorTheme(stateSaver.get().theme),
+  });
+
+  window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", () => {
+    dispatch(actions.osThemeChange());
   });
```
Comment (dsherret) [inline site/src/AppContext.tsx:45]: Nice!

File: shared/src/types.ts (hunk 1 above)
Comment (dsherret) [reply shared/src/types.ts:9]: I wrote this code originally when redux was a thing in order to try it out then switched it to useReducer when upgrading to hooks. I wouldn't use it again tbh, but I'm not really a web developer.

**Review by dsherret — APPROVED** [review]
Great change. Thanks again!

**Pull request merged**